collect_app/src/main/java/org/odk/collect/android/openrosa/okhttp/OkHttpOpenRosaServerClientProvider.java

```diff
@@ -80,6 +88,13 @@ private OkHttpOpenRosaServerClient createNewClient(String scheme, String userAge
                 .readTimeout(READ_CONNECTION_TIMEOUT, TimeUnit.MILLISECONDS)
                 .followRedirects(true);
 
+        if (cacheDir != null) {
+            builder.cache(new Cache(
+                    new File(cacheDir, "http"),
+                    50L * 1024L * 1024L // 50 MiB
```

Comment:
We probably want to experiment with different sizes (or make it configurable), but this seemed good enough for validating that caching works at least.